tests/posix/common/src/mqueue.c

```diff
@@ -97,4 +96,205 @@ ZTEST(mqueue, test_mqueue)
+		      "Error in data reception. exp: %s act: %s", send_data, rec_data);
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+
+	notification_executed = true;
+}
+
+ZTEST(mqueue, test_mqueue_notify_thread)
+{
+	mqd_t mqd;
+	struct mq_attr attrs;
+	struct sigevent not;
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	attrs.mq_msgsize = MESSAGE_SIZE;
+	attrs.mq_maxmsg = MESG_COUNT_PERMQ;
+
+	not.sigev_notify = SIGEV_THREAD;
+	not.sigev_value.sival_int = (int)pthread_self();
+	not.sigev_notify_function = notify_function_thread;
+	not.sigev_notify_attributes = NULL;
+
+	notification_executed = false;
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	mqd = mq_open(queue, flags, mode, &attrs);
+
+	zassert_false(mq_notify(mqd, &not),
+		      "Unable to set notification.");
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
+
+	usleep(USEC_PER_MSEC * 10U);
+
+	zassert_true(notification_executed,
+		     "Notification not triggered.");
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+	zassert_false(mq_unlink(queue), "Not able to unlink Queue");
+}
+
+ZTEST(mqueue, test_mqueue_notify_non_empty_queue)
+{
+	mqd_t mqd;
+	struct mq_attr attrs;
+	struct sigevent not;
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	attrs.mq_msgsize = MESSAGE_SIZE;
+	attrs.mq_maxmsg = MESG_COUNT_PERMQ;
+
+	not.sigev_notify = SIGEV_NONE;
+	not.sigev_value.sival_ptr = (void *)&notification_executed;
+	not.sigev_notify_function = notify_function_basic;
+	not.sigev_notify_attributes = NULL;
+
+	notification_executed = false;
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	mqd = mq_open(queue, flags, mode, &attrs);
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
+
+	zassert_false(mq_notify(mqd, &not),
+		      "Unable to set notification.");
+
+	zassert_false(notification_executed,
+		      "Notification shouldn't be processed.");
+
+	mq_receive(mqd, rec_data, MESSAGE_SIZE, 0);
+	zassert_false(strcmp(rec_data, send_data),
+		      "Error in data reception. exp: %s act: %s", send_data, rec_data);
+
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
+
+	zassert_true(notification_executed,
+		     "Notification not triggered.");
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+	zassert_false(mq_unlink(queue), "Not able to unlink Queue");
+}
+
+ZTEST(mqueue, test_mqueue_notify_errors)
+{
+	mqd_t mqd;
+	struct mq_attr attrs;
+	struct sigevent not;
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	attrs.mq_msgsize = MESSAGE_SIZE;
+	attrs.mq_maxmsg = MESG_COUNT_PERMQ;
+
+	not.sigev_notify_function = notify_function_basic;
+	not.sigev_notify_attributes = NULL;
+	not.sigev_notify = SIGEV_SIGNAL;
+	not.sigev_value.sival_ptr = (void *)&notification_executed;
```

Comment:
Not a blocker, but `attrs` and `not` could be initialized where they are declared (we support C11 by default, so no need to use only C89 declarations). It might be a bit less verbose, but also, there would be no need to initialize anything to 0 / `NULL` explicitly.
```suggestion
	struct mq_attr attrs = {
		.mq_msgsize = MESSAGE_SIZE,
		.mq_maxmsg = MESG_COUNT_PERMQ,
	};
	struct sigevent not = {
		.sigev_notify_function = notify_function_basic,
		.sigev_notify = SIGEV_SIGNAL,
		.sigev_value.sival_ptr = (void *)&notification_executed,
	};
	int32_t mode = 0777;
	int flags = O_RDWR | O_CREAT;
```